lib/index.ts

```diff
@@ -645,14 +698,28 @@ export class Server extends EventEmitter {
     return this;
   }
 
+  public on<
```

Comment:
I suppose we would also need to strongly type `once`, `removeListener`, etc. Perhaps it might make sense to introduce an `abstract class StrictEventEmitter` between `EventEmitter` and `Server`; this would just override all methods to do all the strict typing, and call `super`'s implementations. What do you think of that approach?

Reply:
I ended up moving it all to `StrictEventEmitter` to avoid duplicating all the overrides of `on` and `once`. And I added `once`, which has the exact same signature.